File: app_core/core/src/main/java/com/wangzs/core/network/RxRetrofit.java
package com.wangzs.core.network;



import com.wangzs.core.network.cert.TrustAllManager;
import com.wangzs.core.network.converter.JsonConverterFactory;
import com.wangzs.core.network.interceptor.RxEncryptInterceptor;
import com.wangzs.core.network.interceptor.RxLoggingInterceptor;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;

/**
 * @Description:
 * @Author: wangzs
 * @Date: 2022-03-10
 * @Version:
 */
public final class RxRetrofit {

    private static final int DEFAULT_TIME_OUT = 5;
    private static final int DEFAULT_READ_TIME_OUT = 10;
    private static final int DEFAULT_WRITE_TIME_OUT = 10;

    private Retrofit mRetrofit;

    private static class SingletonHolder {
        private static final RxRetrofit INSTANCE = new RxRetrofit();
    }

    public static RxRetrofit getInstance() {
        return SingletonHolder.INSTANCE;
    }

    private RxRetrofit() {

        OkHttpClient.Builder builder = new OkHttpClient.Builder();
        // time out
        builder.connectTimeout(DEFAULT_TIME_OUT, TimeUnit.SECONDS);
        builder.readTimeout(DEFAULT_READ_TIME_OUT, TimeUnit.SECONDS);
        builder.writeTimeout(DEFAULT_WRITE_TIME_OUT, TimeUnit.SECONDS);
        // cache
//        builder.addNetworkInterceptor(RxCacheInterceptor.onNetwork());
//        builder.addInterceptor(RxCacheInterceptor.notNetwork());
//        builder.cache(RxCacheInterceptor.create());
        // encrypt
//        builder.addInterceptor(RxParamInterceptor.create());
        builder.addInterceptor(RxEncryptInterceptor.create());
        // log
        builder.addInterceptor(RxLoggingInterceptor.create());
        // 证书
        TrustAllManager trustAllManager = new TrustAllManager();
        builder.sslSocketFactory(TrustAllManager.createTrustAllSSLFactory(trustAllManager),
                trustAllManager);
        builder.hostnameVerifier(TrustAllManager.createTrustAllHostnameVerifier());

        mRetrofit = new Retrofit.Builder()
                .client(builder.build())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
//                .addConverterFactory(GsonConverterFactory.create())
                .addConverterFactory(JsonConverterFactory.create())
                .baseUrl(RxUrl.getBaseUrl())
                .build();
    }

    public <T> T create(Class<T> service) {
        return mRetrofit.create(service);
    }
}

File: app_core/core/src/main/java/com/wangzs/core/network/RxRetrofit.kt
package com.wangzs.core.network

import com.wangzs.core.network.cert.TrustAllManager
import com.wangzs.core.network.converter.GsonConverterFactory
import com.wangzs.core.network.converter.JsonConverterFactory
import com.wangzs.core.network.interceptor.RxEncryptInterceptor
import com.wangzs.core.network.interceptor.RxLoggingInterceptor
import okhttp3.OkHttpClient

import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
//import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

/**
 * @Description:
 * @Author: wangzs
 * @Date: 2020-03-10
 * @Version:
 */
object RxRetrofit {

    private const val DEFAULT_TIME_OUT = 5
    private const val DEFAULT_READ_TIME_OUT = 10
    private const val DEFAULT_WRITE_TIME_OUT = 10

    // 证书
    private val trustAllManager = TrustAllManager()
    private val okHttpClient: OkHttpClient = OkHttpClient.Builder()
        .connectTimeout(DEFAULT_TIME_OUT.toLong(), TimeUnit.SECONDS)
        .readTimeout(DEFAULT_READ_TIME_OUT.toLong(), TimeUnit.SECONDS)
        .writeTimeout(DEFAULT_WRITE_TIME_OUT.toLong(), TimeUnit.SECONDS)
//        .addNetworkInterceptor(RxCacheInterceptor.onNetwork())
        .addInterceptor(RxLoggingInterceptor.create())
//        .addInterceptor(
//            RxCacheInterceptor.notNetwork()
//        )
//        .cache(RxCacheInterceptor.create())
//        .addInterceptor(RxParamInterceptor.create())
        .addInterceptor(RxEncryptInterceptor.create())
        .hostnameVerifier(TrustAllManager.createTrustAllHostnameVerifier())
        .sslSocketFactory(
            TrustAllManager.createTrustAllSSLFactory(trustAllManager),
            trustAllManager
        )
        .build()


    private val mRetrofit: Retrofit = Retrofit.Builder()
        .client(okHttpClient)
        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
//        .addConverterFactory(GsonConverterFactory.create())
        .addConverterFactory(JsonConverterFactory.create())
        .baseUrl(RxUrl.getBaseUrl())
        .build()


    fun <T> create(service: Class<T>?): T {
        return mRetrofit.create(service)
    }


}